projects/packages/publicize/src/rest-api/class-connections-controller.php

```diff
@@ -165,23 +177,35 @@ public function get_item_schema() {
 	/**
 	 * Get all connections. Meant to be called directly only on WPCOM.
 	 *
-	 * @param bool $run_tests Whether to run tests on the connections.
+	 * @param array $args Arguments
+	 *                    - 'test_connections': bool Whether to run connection tests.
+	 *                    - 'include': enum('site-wide', 'shared') Which connections to include.
 	 *
 	 * @return array
 	 */
-	protected static function get_all_connections( $run_tests = false ) {
+	protected static function get_all_connections( $args = array() ) {
 		/**
 		 * Publicize instance.
-		 *
-		 * @var \Automattic\Jetpack\Publicize\Publicize $publicize
 		 */
 		global $publicize;
 
 		$items = array();
 
+		$run_tests = $args['test_connections'] ?? false;
+		$include   = $args['include'] ?? '';
+
 		$test_results = $run_tests ? self::get_connections_test_status() : array();
 
-		foreach ( (array) $publicize->get_services( 'connected' ) as $service_name => $connections ) {
+		if ( 'site-wide' === $include ) {
+			$service_connections = $publicize->get_all_connections_for_blog_id( get_current_blog_id() );
+		} else {
+			// `false` means the default behavior, current user.
+			$user_id = 'shared' === $include ? 0 : false;
```

Comment:
Do we not normally return shared connections? Could we maybe swap to that being the default? Would it break anything?

Reply:
Gotcha, but if we changed it so that it did, do you think that would be a problem? Nothing we know of should be using it, and if something is and receives the shared connections I wonder if that would break it 🤔 

I realise it's a potentially breaking change, but adding something is less breaking. I'm not saying we should definitely do it, but it would be great if we could convince ourselves that it's relatively safe and not have to add an extra parameter.

Reply:
Well, I want to make a correction here, the current implementation returns shared connections by default.

So, we can make this a default behavior.